File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String,List> map=new HashMap<>();
        for(String str:strs){
            char[] arr=str.toCharArray();
            Arrays.sort(arr);
            String key=new String(arr);
            List l=map.getOrDefault(key,new ArrayList<String>());
            l.add(str);
            map.put(key,l);
        }
        return new ArrayList(map.values());
    }
}

File: 49-group-anagrams/group-anagrams.py
from collections import defaultdict
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anagrams_map=defaultdict(list)
        for word in strs:
            anagrams_map["".join(sorted(word))].append(word)
        print (anagrams_map)
        return [val for val in anagrams_map.values()]
        